ui/packages/consul-ui/app/services/repository/service-instance.js

```diff
@@ -34,44 +32,6 @@ export default class ServiceInstanceService extends RepositoryService {
 
   @dataSource('/:partition/:ns/:dc/service-instance/:serviceId/:node/:id')
   async findBySlug(params, configuration = {}) {
-    if (typeof configuration.cursor !== 'undefined') {
-      params.index = configuration.cursor;
-      params.uri = configuration.uri;
-    }
-    return this.authorizeBySlug(
-      async () => this.store.queryRecord(this.getModelName(), params),
-      ACCESS_READ,
-      params
-    );
-  }
-
-  @dataSource('/:partition/:ns/:dc/proxy-service-instance/:serviceId/:node/:id')
-  async findProxyBySlug(params, configuration = {}) {
-    const instance = await this.findBySlug(...arguments);
-    let proxy = this.store.peekRecord('proxy', instance.uid);
-    // Currently, we call the proxy endpoint before this endpoint
-    // therefore proxy is never undefined. If we ever call this endpoint
-    // first we'll need to do something like the following
-    // if(typeof proxy === 'undefined') {
-    //   await proxyRepo.create({})
-    // }
-
-    // Copy over all the things to the ProxyServiceInstance
-    ['Service', 'Node', 'meta'].forEach(prop => {
-      set(proxy, prop, instance[prop]);
-    });
-    ['Checks'].forEach(prop => {
-      // completely wipe out any previous values so we don't accumulate things
-      // eternally
-      proxy.set(prop, []);
-      instance[prop].forEach(item => {
-        if (typeof item !== 'undefined') {
-          proxy[prop].addFragment(item.copy());
-        }
-      });
-    });
-    // delete the ServiceInstance record as we now have a ProxyServiceInstance
-    instance.unloadRecord();
-    return proxy;
```

Comment:
This turned out to be the problematic code.

When we request details on a service we use a proxy endpoint to check if it has a proxy service, and if it does we use the previous same service endpoint to get the details for the proxy. But! The extra bit we did here was to munge together the details we got from the proxy endpoint and the details we get from the service endpoint together to make a special type of 'ProxyServiceInstance' - a special type of ServiceInstance.

Turns out, doing this was providing very little benefit to us. Moreover previously we could almost guarantee that the proxy request would always be called before the service request (almost in the fact that there was probably a very very edge case bug here in doing this anyway that has just never surfaced as its unlikely to happen).

Now its pretty common that the ServiceProxyInstance has been created and munged together, and then we check the details for the Proxy again and the `instance` in the above code gets 'refreshed/updated' again, but without then re-requesting and re-munging the extra details onto the instance, meaning a bunch of properties that we weere expecting to be set have been deleted/set back to undefined.

All in all its a little complex to explain well so would be a good one to chat through if you want the overall picture a little more clearly.

Alternatively - we just deleted the entire thing and used a different approach that doesn't have these issues, by using less EmberData and Fragments.